sympy/solvers/solveset.py

```diff
@@ -1705,6 +1707,14 @@ def linear_eq_to_matrix(equations, *symbols):
     if hasattr(symbols[0], '__iter__'):
         symbols = symbols[0]
 
+    # setting default value for check_linearity
+    if 'check_linearity' not in options:
+        options['check_linearity'] = False
+
+    if options['check_linearity']:
+        # removing duplicates from symbols
+        symbols = list(set(symbols))
```

Comment:
Seems like this should always be done but you should retain the order of the passed symbols. `symbols = uniq(symbols)` would be preferred (with `uniq` coming from sympy.utilities.iterables).

Reply:
I have implemented it in a different way. Please check and give your suggestions for improvements.